charts/kvisor/values.yaml

```diff
@@ -60,23 +60,15 @@ agent:
   # Allow to set priority class like system-node-critical.
   priorityClass: ""
 
-  podAnnotations: { }
+  podAnnotations: {}
 
-  podSecurityContext: { }
-  # fsGroup: 2000
+  securityContext: {}
 
   containerSecurityContext:
     privileged: true
```

Comment:
It seems not possible to run without privileged, I tried with the following capabilities:

```yaml
# inspired from https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/values.yaml#L281-L350
securityContext:
  capabilities:
    add:
    - CHOWN
    - KILL
    - NET_ADMIN
    - NET_RAW
    - IPC_LOCK
    - SYS_MODULE
    - SYS_ADMIN
    - SYS_RESOURCE
    - DAC_OVERRIDE
    - FOWNER
    - SETGID
    - SETUID
    - SYS_CHROOT
    - SYS_PTRACE
    - SYS_RESOURCE
    drop:
    - ALL
 ```

And the agent fails at startup with the following errors:

```shelll
# if process tree exporter is enabled
time=2024-09-30T17:12:17.527Z level=ERROR msg="initialize process tree: open 3591/root/proc: permission denied"

# if netflow exporter is enabled
time=2024-09-30T17:15:48.291Z level=ERROR msg="conntrack: permission denied"
```

Reply:
The process one requires root user for sure 🤔 
So it should be unrelated to privileged 🤔

Reply:
Ok after some tracing of how `procfs` works, I guess (at least `process_tree`) should work by setting `CAP_SYS_PTRACE`. From what I understand, you need to either have the same creds as the process represented as the `/proc/<pid>` entry, of have `CAP_SYS_PTRACE`.

Reply:
Ok I see you have `SYS_PTRACE` in the capabilities you tried hmm.

Reply:
apparmor was preventing to read from `/proc/<pid>`